html/modules/custom/reliefweb_job_tagger/src/Plugin/QueueWorker/OchaAiJobTagTaggerWorker.php

```diff
@@ -573,4 +607,59 @@ public function getMostRelevantTermsFromEs(
     return $vocabularies;
   }
 
+  /**
+   * Get similar jobs.
+   */
+  protected function getSimilarJobs(NodeInterface $node, string $field) {
+    $nid = $node->id();
+    $relevant = $this->jobTagger->getSimilarDocuments($nid, $node->get('body')->value);
+    if (empty($relevant)) {
+      return [];
+    }
+
+    $max = reset($relevant);
+
+    /** @var \Drupal\node\Entity\Node[] $nodes */
+    $nodes = $this->entityTypeManager->getStorage('node')->loadMultiple(array_keys($relevant));
+
+    if (isset($nodes[$nid])) {
+      unset($nodes[$nid]);
+    }
+
+    $terms = [];
+    foreach ($nodes as $node) {
+      if ($node->hasField($field) && !$node->get($field)->isEmpty()) {
```

Comment:
This doesn't take into account the fact that jobs can be tagged with multiple themes.

Reply:
Loop added